### ISSUE TITLE
agent crash on FIM

### ISSUE THREAD

**Issue by rulezz55** [wazuh/wazuh#6263]
| Wazuh version | Component | Install type | Install method | Platform |
| --- | --- | --- | --- | --- |
| 3.13.2-1 | Syscheckd | Agent | Packages | Windows server 2016 |

Hello! There is a file server with 2,000,000 files, I ran into the syscheck limitation by default there is a possibility to monitor 100,000 files.
to bypass the restriction in
ossec.conf: added file_limit to syscheck section:
```
  <syscheck>
    <disabled> no </disabled>
    <frequency> 43200 </frequency>
    <directories check_all = "yes" whodata = "yes"> D: / </directories>
<file_limit>
  <enabled> yes </enabled>
  <entries> 2000000 </entries>
</file_limit>
  </syscheck>
```
 The agent processed 400,000 files and exited with a fatal error
```
fim_db.c: 918 at fim_db_insert (): DEBUG: Couldn't insert 'path to file' entry into DB. The DB is full, please check your configuration.
2020/10/11 18:06:28 ossec-agent [980] win_service.c: 256 at OssecServiceCtrlHandler (): INFO: Received exit signal.
2020/10/11 18:06:28 ossec-agent [980] win_service.c: 258 at OssecServiceCtrlHandler (): INFO: Exiting ...
2020/10/11 18:06:28 ossec-agent [980] log_builder.c: 226 at log_builder_update_hostname (): CRITICAL: At pthread_rwlock_wrlock (): Invalid argument

```
Tell me what is the reason for the default limit of 100,000? How to overcome this limitation and monitor 2,000,000 files?

**Comment by jesusjimsa** [comment]
Hello, @rulezz55.

About the reason for the limitation being 100,000, it was the average number of files that users usually have monitored.

Regarding your problem, I have been trying to reproduce your issue but I haven't been able to do it. While I investigate this problem, if you have 2,000,000 files in this file server and you want to monitor all of them, you could try disabling the `file_limit` option:

```xml
<file_limit>
  <enabled>no</enabled>
  <entries>100000</entries>
</file_limit>
```

I will keep trying to reproduce this error and I will get back to you when I find the problem causing Wazuh to stop working.

**Comment by vikman90** [comment]
@rulezz55 @jesusjimsa just following up on this issue.

As said, you should be able to monitor as many files as you need, taking into account that the greater the number of files, the greater the consumption of resources.

Still trying to reproduce this problem. I miss the date-time part of the first log, in order to know whether the agent crashes immediately after filling up the database. Both logs are related to different components (FIM and Logcollector). I wonder if a bug in FIM is impacting other components.

However, we've introduced a bunch of fixes for FIM in 4.0 ([see the changelog](https://github.com/wazuh/wazuh/blob/v4.0.0/CHANGELOG.md#fixed)). Did you have the chance to upgrade your agent? Hopefully, the bug could disappear. Otherwise, please let us know if this problem appears always or intermittently.

Thank you for reporting this issue.
Best regards.

**Comment by vikman90** [comment]
Closing this issue due to inactivity. 
Feel free to reopen it if you detect this issue in 4.0.

Thank you.
